src/include/miopen/find_solution.hpp

```diff
@@ -82,6 +82,17 @@ auto FindSolutionImpl(
                 MIOPEN_LOG_WE("Invalid config loaded from Perf Db: "
                               << s.SolverDbId() << ": " << config << ". Performance may degrade.");
             }
+            else if(!s.AltSolverDbId().empty() && db.Load(context, s.AltSolverDbId(), config))
+            {
+                MIOPEN_LOG_I("Perf Db: alternate record loaded: " << s.AltSolverDbId());
+                if(s.IsValidPerformanceConfig(context, config))
+                {
+                    return s.GetSolution(context, config);
+                }
+                MIOPEN_LOG_WE("Invalid alternate record loaded from Perf Db: "
+                              << s.AltSolverDbId() << ": " << config
+                              << ". Performance may degrade.");
+            }
```

Comment:
@JehandadKhan You may also generalize this code do avoid duplication for first and second DB loading, but I don't insist.